test/module/irohad/torii/processor/consensus_status_processor_test.cpp

```diff
@@ -0,0 +1,284 @@
+ * communication service
+ * @and   all transactions composed the block @and were committed
+ * @then for every transaction COMMIT status is returned
+ */
+TEST_F(ConsensusStatusProcessorTest, TransactionProcessorOnCommitTest) {
+  std::vector<shared_model::proto::Transaction> txs;
+  for (size_t i = 0; i < proposal_size; i++) {
+    auto &&tx = addSignaturesFromKeyPairs(baseTestTx(), makeKey());
+    txs.push_back(tx);
+  }
+
+  EXPECT_CALL(*status_bus, publish(_)).Times(txs.size() * 2);
+
+  EXPECT_CALL(*status_factory, makeStatefulValid(_, _)).Times(txs.size());
+  EXPECT_CALL(*status_factory, makeCommitted(_, _)).Times(txs.size());
+
+  // 1. Create proposal and notify transaction processor about it
+  auto proposal = std::make_shared<shared_model::proto::Proposal>(
+      TestProposalBuilder().transactions(txs).build());
+
+  // empty transactions errors - all txs are valid
+  verified_prop_notifier.get_subscriber().on_next(
+      std::make_shared<iroha::validation::VerifiedProposalAndErrors>(
+          std::make_pair(proposal, iroha::validation::TransactionsErrors{})));
+
+  auto block = TestBlockBuilder().transactions(txs).build();
+
+  // 2. Create block and notify transaction processor about it
+  SynchronizationEvent commit_event{
+      rxcpp::observable<>::just(
+          std::shared_ptr<shared_model::interface::Block>(clone(block))),
+      SynchronizationOutcomeType::kCommit};
+  commit_notifier.get_subscriber().on_next(commit_event);
+}
+
+/**
+ * @given cs processor
+ * @when  some transactions became part of block, while some
+ * were not committed, failing stateful validation
+ * @then  for every transaction from block COMMIT status is returned
+ * @and   for every transaction, which failed stateful validation,
+ * STATEFUL_INVALID_STATUS status is returned
+ */
+TEST_F(ConsensusStatusProcessorTest, TransactionProcessorInvalidTxsTest) {
+  std::vector<shared_model::proto::Transaction> block_txs;
+  for (size_t i = 0; i < block_size; i++) {
+    auto &&tx = TestTransactionBuilder().createdTime(i).build();
+    block_txs.push_back(tx);
+    status_map[tx.hash()] = external_status_factory->makeNotReceived(
+        tx.hash(),
+        shared_model::interface::TxStatusFactory::emptyErrorMessage());
+  }
+
+  std::vector<shared_model::proto::Transaction> invalid_txs;
+
+  for (size_t i = block_size; i < proposal_size; i++) {
+    auto &&tx = TestTransactionBuilder().createdTime(i).build();
+    invalid_txs.push_back(tx);
+    status_map[tx.hash()] = external_status_factory->makeNotReceived(
+        tx.hash(),
+        shared_model::interface::TxStatusFactory::emptyErrorMessage());
+  }
+
+  // For all transactions from proposal
+  // transaction will be published twice
+  // (first that they are stateless
+  // valid and second that they either
+  // passed or not stateful validation)
+  // Plus all transactions from block will
+  // be committed and corresponding status will be sent
```

Comment:
Possible paraphrasing.

Each transaction from proposal will be published twice.
The first time - stateless valid status, the second - passed or stateful invalid status.
Additionally, all the transactions from a block will be committed and corresponding statuses will be sent.